### DISCUSSION TITLE
[Problem] How Use Shell Like Telegram APP?

### DISCUSSION THREAD

**Discussion by Saleh-Software** [xamarin/Xamarin.Forms#11804] [General]
[Telegram Tab](http://uupload.ir/files/0dgz_photo_2020-08-16_20-06-04.jpg)

i want like this design but xamarin hide the photo in tabs!

tnx

**Comment by Saleh-Software** [comment]
any example code?

**Comment by jsuarezruiz** [comment]
@DenisHidden Do you mean to be able to add badges, etc?

**Comment by Saleh-Software** [comment]
hi i want change icon like telegram on tab
profile tab u can see 

how like telegram we can made it?

**Comment by Saleh-Software** [comment]
when we put image color will be gone 

like gray and light blue i mean how show real pic...

**Reply by davidortinau** [reply]
@DenisHidden check out this app that uses styles and triggers to change the tab icons in a shell app at runtime. https://github.com/xamcat/xam-twitch/blob/master/XamTwitch/XamTwitch/AppShell.xaml#L30

In this case it's a `FontImageSource`, but could be any valid type. The concept could be extended for custom scenarios if this is close to what you need.

**Reply by davidortinau** [reply]
We also have added an `IsVisible` bindable prop that might be useful if you want tabs visible sometimes but not others.

**Reply by davidortinau** [reply]
Or (sorry keep thinking of different things you might do) you could describe multiple sets of `Tab`s within different `TabBar`s that have distinct routes, and you could navigate to the one that matches your scenario. Since the pages are all templates, you're really just configuring didn't "looks" for your app navigation menu (tabs or flyout or both).

The first `TabBar` or `FlyoutItem` (aka `ShellItem`) will be displayed first when your app loads. The other sets of tabs or navigation will not appear until you navigate to them by `Route`.
